gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+
+        logger.warning(f"GunicornApplicationSeparator: matched against process {process} but couldn't find WSGI module")
+        return None
+
+
+class UwsgiApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "uwsgi"
+
+    def is_supported(self, process: Process) -> bool:
+        return process.cmdline()[0] == "uwsgi"
+
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        wsgi = get_positional_arg(process.cmdline(), "-w")
+        if wsgi is not None:
+            return wsgi
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return wsgi
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
+        return None
+
+
+class CeleryApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "celery"
+
+    def is_supported(self, process: Process) -> bool:
+        if process.cmdline()[0] == "celery":
+            return True
+
+        return len(process.cmdline()) >= 3 and ["-m", "celery"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        app_name = get_positional_arg(process.cmdline(), "-A") or get_positional_arg(process.cmdline(), "--app")
+        if app_name is None:
+            logger.warning("Couldn't find positional argument -A or --app for application indication")
+            return None
+
+        return app_name
+
+
+class PySparkApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "pyspark"
+
+    def is_supported(self, process: Process) -> bool:
+        # We're looking for pythonXX -m pyspark.daemon
+        return process.cmdline() >= 3 and ["-m", "pyspark.daemon"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        return "generic pyspark.daemon"
+
+
+class PythonModuleApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "generic-python"
+
+    def is_supported(self, process: Process) -> bool:
+        if "python" not in process.cmdline()[0]:
```

Comment:
Better to match against `r"^python([23](\.\d{1,2})?)?$"`

Reply:
nope, since it can be some `/usr/bin/python` without a number, actually it's very common (many applications are being run from just `python` executable -> in most of the cases this executable is a symlink to an executable with some version identifier in it's name)

Reply:
This regex will match `python`, `python3` and `python3.10`.
On the other hand, if a client will decide to run an application named "happythongs" for any reason, this condition specifically would match it too.
BTW, the suggested regex won't match against execution of Python with a full path (i.e. if you run `/use/bin/python` directly), but it can be expanded.

Reply:
kk changed

Reply:
Done, resolving, the regex seems okay now:
```
In [2]: _PYTHON_BIN_RE = re.compile(r"^python([23](\.\d{1,2})?)?$")                                                                                                                                        

In [3]: _PYTHON_BIN_RE.match("python")                                                                                                                                                                     
Out[3]: <re.Match object; span=(0, 6), match='python'>

In [4]: _PYTHON_BIN_RE.match("python2")                                                                                                                                                                    
Out[4]: <re.Match object; span=(0, 7), match='python2'>

In [5]: _PYTHON_BIN_RE.match("python2.7")                                                                                                                                                                  
Out[5]: <re.Match object; span=(0, 9), match='python2.7'>

In [6]: _PYTHON_BIN_RE.match("python2.76")                                                                                                                                                                 
Out[6]: <re.Match object; span=(0, 10), match='python2.76'>
```